fontir/src/ir.rs

```diff
@@ -553,6 +551,48 @@ impl GlobalMetrics {
         }
     }
 
+    pub fn build(self, axes: &Axes) -> Result<GlobalMetrics, Error> {
```

Comment:
- I would probably rename `set` above to something more builder-y, like 'add'?
- I would document the `build` method, and in particular I would describe the circumstances in which it will return an error.

Reply:
For the former: I've now documented the `build` function, and the main error cases to consider.

For the latter: I've left this as `set` to reflect that you can clobber existing values with it, but updated the docs in kind; I tried out the alternative name as normally `add` shouts build to me more too, but along with the previous concern it also made the patch a lot heavier, and so I think we should be fine to keep it as-is if we're happy to.